File: frontend/src/app/core/services/admin/devise.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';

export interface DeviseRequest {
  code: string;
  label: string;
  taux: number;
  active: boolean;
}

export interface DeviseResponse {
  id: number;
  code: string;
  label: string;
  taux: number;
  active: boolean;
}

@Injectable({ providedIn: 'root' })
export class DeviseService {
  private baseUrl = '/api/v1/admin/devises';

  constructor(private http: HttpClient) {}

  findAll(): Observable<DeviseResponse[]> {
    return this.http.get<DeviseResponse[]>(this.baseUrl);
  }

  save(devise: DeviseRequest): Observable<DeviseResponse> {
    return this.http.post<DeviseResponse>(this.baseUrl, devise);
  }

  update(id: number, devise: DeviseRequest): Observable<DeviseResponse> {
    return this.http.put<DeviseResponse>(`${this.baseUrl}/${id}`, devise);
  }

  delete(id: number): Observable<void> {
    return this.http.delete<void>(`${this.baseUrl}/${id}`);
  }
}

File: backend/src/main/java/com/ebanking/core/service/devise/DeviseService.java
package com.ebanking.core.service.devise;


import com.ebanking.core.domain.base.devise.Devise;
import com.ebanking.core.dto.devise.DeviseRequest;
import com.ebanking.core.dto.devise.DeviseResponse;

import java.util.List;
import java.util.Optional;

public interface DeviseService {
    List<DeviseResponse> findAll();
    DeviseResponse findById(Long id);
    DeviseResponse save(DeviseRequest request);
    DeviseResponse update(Long id, DeviseRequest request);
    void delete(Long id);
}
